plugins/org.locationtech.udig.catalog/src/org/locationtech/udig/catalog/ICatalog.java

```diff
@@ -118,115 +118,111 @@ public String toString() {
-            throws IllegalArgumentException;
-
+    public abstract IGeoResource createTemporaryResource( Object descriptor ) throws IllegalArgumentException;
+    
     /**
+     * Returns The list of class names that this catalog can use to create Temporary Resources.
      *
      * @return The list of class names that this catalog can use to create Temporary Resources.
      */
     public abstract String[] getTemporaryDescriptorClasses();
+    
 
-    @Override
     public void dispose(IProgressMonitor monitor) {
-        monitor.beginTask(Messages.ICatalog_dispose, 100);
-        List<? extends IResolve> members;
-        try {
-            members = members(SubMonitor.convert(monitor, 1));
-        } catch (Throwable e) {
-            ErrorManager.get().displayException(e,
-                    "Error disposing members of catalog: " + getIdentifier(), CatalogPlugin.ID); //$NON-NLS-1$
-            return;
-        }
-        int steps = (int) ((double) 99 / (double) members.size());
-        for (IResolve resolve : members) {
+            monitor.beginTask(Messages.ICatalog_dispose, 100);
+            List< ? extends IResolve> members;
             try {
-                SubMonitor subMonitor = SubMonitor.convert(monitor, steps);
-                resolve.dispose(subMonitor);
-                subMonitor.done();
+                members = members(new SubProgressMonitor(monitor,1) );
             } catch (Throwable e) {
-                ErrorManager.get().displayException(e,
-                        "Error disposing members of catalog: " + getIdentifier(), CatalogPlugin.ID); //$NON-NLS-1$
+                ErrorManager.get().displayException(e, "Error disposing members of catalog: "+getIdentifier(), CatalogPlugin.ID); //$NON-NLS-1$
+                return;
             }
-        }
-
+            int steps=(int)((double)99/(double)members.size());
+            for( IResolve resolve : members ) {
+                try {
+                    SubProgressMonitor subProgressMonitor = new SubProgressMonitor(monitor, steps);
+                    resolve.dispose( subProgressMonitor);
+                    subProgressMonitor.done();
+                } catch (Throwable e) {
+                    ErrorManager.get().displayException(e, "Error disposing members of catalog: "+getIdentifier(), CatalogPlugin.ID); //$NON-NLS-1$
+                }
+            }
+        
     }
-
+    
+    
     /**
-     * This method evaluates the results of constructServices and returns the list of Services that
+     * This method evaualtes the results of constructServices and returns the list of Services that
      * have already been added to the catalog.
      * <p>
      * Services that have already been added to the catalog will be properly disposed when the
-     * application shuts down; you do not need to call dispose on these services yourself.
-     *
+     * applicaiton shuts down; you do not need to call dispose on these services yourself.
```

Comment:
Hmm, what a language have you configured in your workspace?

Reply:
I reverted one commit so this file is not changed within this PR at all.